src/System.Net.Http/src/System/Net/Http/HttpClient.cs

```diff
@@ -394,7 +383,17 @@ public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, HttpCompl
             }
             finally
             {
-                linkedCts.Dispose();
+                try
+                {
+                    // When a request completes, dispose the request content so the user doesn't have to. This also
+                    // helps ensure that a HttpContent object is only sent once using HttpClient (similar to HttpRequestMessages
+                    // that can also be sent only once).
+                    request.Content?.Dispose();
+                }
+                finally
+                {
+                    linkedCts.Dispose();
+                }
             }
```

Comment:
> Is the extra try/finally needed here? Is it expected that Dispose will throw?

I don't know, I was just being cautious and preserving the existing semantics.

> is it really bad for the CTS to not be disposed?

It depends on whether the user passed in a long-lived CancellationToken.  Not disposing of the linked token source will leave some state registered in that CancellationToken, so if that token sticks around for a while, it could end up being a long-lived leak.